src/UILayer/web/src/__tests__/e2e/auth-flow.test.tsx

```
@@ -0,0 +1,202 @@
+  let token: string | null = null
+  let user: { name: string; roles: string[] } | null = null
+  const listeners = new Set<() => void>()
+
+  return {
+    getToken: () => token,
+    getUser: () => user,
+    login: (t: string, u: { name: string; roles: string[] }) => {
+      token = t
+      user = u
+      localStorage.setItem("auth_token", t)
+      listeners.forEach((l) => l())
```

Biome (2.4.6)

[error] This callback passed to forEach() iterable method should not return a value.

(lint/suspicious/useIterableCallbackReturn)